nunaliit2-couch-metadata/src/main/java/ca/carleton/gcrc/couch/metadata/IndexServlet.java

```diff
@@ -0,0 +1,240 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(IndexServlet.class);
+
+    /**
+     * Regex to parse module document Id from query string.
+     */
+    private static final Pattern URL_PATTERN = Pattern.compile("module=([^#]+)#?.*");
+
+    /**
+     * Regex matcher used to find module Id in query string.
+     */
+    private static final Matcher matcher = URL_PATTERN.matcher("");
+
+    private CouchDb couchDb;
+    private HtmlAttachmentChangeListener indexDbChangeListener;
+    private ModuleMetadataChangeListener moduleMetadataChangeListener;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        logger.info("Initialization started");
+
+        ServletContext context = config.getServletContext();
+
+        Object obj = context.getAttribute(CONFIG_DOCUMENT_DB);
+        if (obj == null) {
+            logger.error("Document database is not specified: {}", CONFIG_DOCUMENT_DB);
+            throw new ServletException(String.format("Document database is not specified (%s)", CONFIG_DOCUMENT_DB));
+        }
+        else if (obj instanceof CouchDb) {
+            couchDb = (CouchDb) obj;
+        }
+        else {
+            throw new ServletException("Unexpected object type for document database: " + obj.getClass().getName());
+        }
+
+        Object dbListener = context.getAttribute(INDEX_DB_CHANGE_LISTENER);
+        if (dbListener == null) {
+            logger.error("Index DB change listener is not specified: {}", INDEX_DB_CHANGE_LISTENER);
+            throw new ServletException(String.format("Index DB change listener is not specified (%s)", INDEX_DB_CHANGE_LISTENER));
+        }
+        else if (dbListener instanceof HtmlAttachmentChangeListener) {
+            indexDbChangeListener = (HtmlAttachmentChangeListener) dbListener;
+        }
+        else {
+            throw new ServletException("Unexpected object type for index DB change listener: " + dbListener.getClass().getName());
+        }
+
+        Object metadataListener = context.getAttribute(MODULE_METADATA_CHANGE_LISTENER);
+        if (metadataListener == null) {
+            logger.error("Module metadata change listener is not specified: {}", MODULE_METADATA_CHANGE_LISTENER);
+            throw new ServletException(String.format("Module metadata change listener is not specified (%s)", MODULE_METADATA_CHANGE_LISTENER));
+        }
+        else if (metadataListener instanceof ModuleMetadataChangeListener) {
+            moduleMetadataChangeListener = (ModuleMetadataChangeListener) metadataListener;
+        }
+        else {
+            throw new ServletException("Unexpected object type for module metadata change listener: " + metadataListener.getClass().getName());
+        }
+
+        logger.info("Initialization finished");
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
```

Comment:
Is this paragraph html element suppose to be inside this multi-line comment?

Reply:
Supported per JavaDoc spec.